### PR TITLE
Ported Laser Pointers back into code

### PR DESCRIPTION
**Pull request by dragoslava**
Re-Adds Laser Pointers from old codebase

### CODE REVIEW

**Comment by BotBOREALIS** [comment]
Icons with diff:
<details><summary>icons/obj/projectiles.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
red_laser||red_laser|Created
red_laser[1]||![red_laser[1]](http://images.aurorastation.org/images/70fd1f36113380aba1167c8f5e551661.png)|Created
blue_laser||blue_laser|Created
blue_laser[1]||![blue_laser[1]](http://images.aurorastation.org/images/5b9755e8d771acc9083ab3d32f9b9033.png)|Created
green_laser||green_laser|Created
green_laser[1]||![green_laser[1]](http://images.aurorastation.org/images/1b8c06da7258037ba5889224c0b0add0.png)|Created
purple_laser||purple_laser|Created
purple_laser[1]||![purple_laser[1]](http://images.aurorastation.org/images/648643054767c80ff2e195c08264f15f.png)|Created
</details>

File: code/game/objects/items/devices/laserpointer.dm [hunk 1]
```diff
@@ -0,0 +1,126 @@
+/obj/item/weapon/gun/energy/laser_pointer
+	name = "laser pointer"
+	desc = "Don't shine it in your eyes!"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "pointer"
+	item_state = "pen"
+	var/pointer_icon_state
+	slot_flags = SLOT_BELT
+	w_class = 2 //Increased to 2, because diodes are w_class 2. Conservation of matter.
+	origin_tech = "combat=1"
```
Comment (Alberyk) [inline code/game/objects/items/devices/laserpointer.dm]: That is how this variable is used, also, there are two of them.

File: code/game/objects/items/devices/laserpointer.dm [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	origin_tech = "magnets=2"
+	var/turf/pointer_loc
+	var/energy = 15
+	var/max_energy = 15
+	self_recharge = 1
+	recharge_time = 5
+	var/obj/item/weapon/stock_parts/micro_laser/diode //used for upgrading!
```
Comment (Alberyk) [inline code/game/objects/items/devices/laserpointer.dm]: Still, replacing the micro laser changes nothing at all...?

File: code/game/objects/items/devices/laserpointer.dm [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+
+/obj/item/weapon/gun/energy/laser_pointer/attack(mob/living/M, mob/user)
+	laser_act(M, user)
+
+/obj/item/weapon/gun/energy/laser_pointer/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/weapon/stock_parts/micro_laser))
+		if(!diode)
+			user.drop_item()
+			W.loc = src
```
Comment (Alberyk) [inline code/game/objects/items/devices/laserpointer.dm]: Use force_move.

File: code/game/objects/items/devices/laserpointer.dm [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+			diode = W
+			user << "<span class='notice'>You install a [diode.name] in [src].</span>"
+		else
+			user << "<span class='notice'>[src] already has a cell.</span>"
+
+	else if(istype(W, /obj/item/weapon/screwdriver))
+		if(diode)
+			user << "<span class='notice'>You remove the [diode.name] from the [src].</span>"
+			diode.loc = get_turf(src.loc)
```
Comment (Alberyk) [inline code/game/objects/items/devices/laserpointer.dm]: Use force_move.

File: code/game/objects/items/devices/laserpointer.dm [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+		outmsg = "<span class='notice'>You hit the lens of [C] with [src], temporarily disabling the camera!</span>"
+
+		msg_admin_attack("\[[time_stamp()]\] [user.name] ([user.ckey]) EMPd a camera with a laser pointer")
+		user.attack_log += text("\[[time_stamp()]\] [user.name] ([user.ckey]) EMPd a camera with a laser pointer")
+
+	if(iscarbon(target))
+		if(user.zone_sel.selecting == "eyes")
+			var/mob/living/carbon/C = target
+
+			C.eye_blind = 3
```
Comment (Alberyk) [inline code/game/objects/items/devices/laserpointer.dm]: This does not check for any kind of eye of face protection, makes little sense.

File: code/game/objects/items/devices/laserpointer.dm [hunk 6]
```diff
@@ -0,0 +1,126 @@
+/obj/item/weapon/gun/energy/laser_pointer
+	name = "laser pointer"
+	desc = "Don't shine it in your eyes!"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "pointer"
```
Comment (Alberyk) [inline code/game/objects/items/devices/laserpointer.dm]: No sprites as well.

Comment (dragoslava) [reply code/game/objects/items/devices/laserpointer.dm]: the sprites were never removed from code

https://i.imgur.com/6JHvVT9.jpg

File: code/game/objects/items/devices/laserpointer.dm [hunk 7]
```diff
@@ -0,0 +1,121 @@
+
+
+/obj/item/device/laser_pointer
+	name = "laser pointer"
+	desc = "Don't shine it in your eyes!"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "pointer"
+	item_state = "pen"
+	var/pointer_icon_state
```
Comment (Arrow768) [inline code/game/objects/items/devices/laserpointer.dm]: There should be a sensible default icon state

File: code/game/objects/items/devices/laserpointer.dm [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+	slot_flags = SLOT_BELT
+	w_class = 1
+	var/turf/pointer_loc
+	var/obj/item/weapon/stock_parts/micro_laser/diode //cant use the laser without it
+
+
+
+
+/obj/item/device/laser_pointer/red
+	pointer_icon_state = "red_laser"
```
Comment (Arrow768) [inline code/game/objects/items/devices/laserpointer.dm]: Turn that into the default icon state and eliminate the /red object

File: code/game/objects/items/devices/laserpointer.dm [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+/obj/item/device/laser_pointer/blue
+	pointer_icon_state = "blue_laser"
+/obj/item/device/laser_pointer/purple
+	pointer_icon_state = "purple_laser"
+
+
+/obj/item/device/laser_pointer/Initialize()
+	icon_state = "pointer"
+
+/obj/item/device/laser_pointer/New()
```
Comment (Arrow768) [inline code/game/objects/items/devices/laserpointer.dm]: Dont use new, use initialize

File: code/game/objects/items/devices/laserpointer.dm [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+	..()
+	diode = new(src)
+	icon_state = "pointer"
+	if(!pointer_icon_state)
+		pointer_icon_state = pick("red_laser","green_laser","blue_laser","purple_laser")
+
+/obj/item/device/laser_pointer/upgraded/New()
```
Comment (Arrow768) [inline code/game/objects/items/devices/laserpointer.dm]: Dont use new. use initialize

**Comment by skull132** [comment]
Marked as stale, closing.

Reopen whenever you get around to updating the PR.

**Pull request closed without merging**